Civi/Share/CiviMRF/Core.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Civi\Share\CiviMRF;
+
+use Civi\Api4\ShareNodePeering;
+use CMRF\Core\Connection;
+use CMRF\Exception\ProfileNotFoundException;
+use CMRF\PersistenceLayer\CallFactory;
+
+class Core extends \CMRF\Core\Core {
+
+  public function __construct() {
+    parent::__construct(new CallFactory(
+      ['\Civi\Share\CiviMRF\Call', 'createNew'],
+      ['\Civi\Share\CiviMRF\Call', 'createWithRecord']
+    ));
+  }
+
+  protected function getConnection($connector_id): Connection {
+    return new \CMRF\Connection\CurlAuthX($this, $connector_id);
+  }
+
+  public function getConnectionProfiles() {
+    return Profile::loadMultiple();
+  }
+
+  public function getDefaultProfile() {
+    // CiviShare does not support default profiles.
+    return NULL;
+  }
+
+  protected function getRegisteredConnectors() {
+    return Connector::loadMultiple();
+  }
+
+  public function registerConnector($connector_id) {
+    // Not supported. Connectors are represented by ShareNodePeering entities.
```

Comment:
Maybe throw an exception instead? (Same holds true for the methods below.)

Reply:
not needed for the POC phase.